controller-client/src/lib.rs

```diff
@@ -325,9 +327,24 @@ pub trait ControllerClient: Send + Sync {
     async fn check_scale(&self, stream: &ScopedStream, scale_epoch: i32) -> ResultRetry<bool>;
 }
 
+#[derive(Clone)]
+struct AuthInterceptor {
+    token: Option<String>,
+}
+
+impl Interceptor for AuthInterceptor {
```

Comment:
Could you elaborate more on this change? I think it is unrelated to the PR.

Reply:
Yes, it's the necessary change for upgrading Tonic version. I am following the example [here](https://github.com/hyperium/tonic/blob/master/examples/src/interceptor/client.rs#L45)